### ISSUE TITLE
names of speakers

### ISSUE THREAD

**Issue by ryanshrott** [m-bain/whisperX#195]
is it possible to use examples of speakers to label the speakers names when using --diarize?

**Comment by thomasmol** [comment]
I don't think that is possible, the embedding models for diarization don't know what name/label belongs to who, just that there are different speakers. So labels of ```Speaker n``` are assigned. Maybe it would be possible in the future for the model to infer names (or some label like 'interviewer' and 'interviewee') by looking at the transcript?

**Comment by sorgfresser** [comment]
Another option is using SpeakerRecognition, but this would at least require sample wavs. If you're interested in that, you might want to take a look at [this model](https://huggingface.co/speechbrain/spkrec-ecapa-voxceleb) here. But surely there are other ones at huggingface too.

**Comment by caryknoop** [comment]
As a workaround, you could merge the to be transcribed wav file with a voiceprint wav with known speakers so you know what SPEAKER_? is assigned to the voiceprints and then map that onto the rest of the SPEAKERS. Of course you would need to recalculate the timings and subtract the duration of the voiceprints.   

However, this method does not overcome the real limitations and mistakes of the current diarization process.

**Comment by Navodplayer1** [comment]
@ryanshrott @thomasmol @sorgfresser @caryknoop 
you can use python package created by me: https://pypi.org/project/speechlib/
This is Github repo: https://github.com/Navodplayer1/speechlib

this package can do transcription, speaker diarization and speaker recognition all together and give a transcript with actual speaker names!

**Comment by Barabazs** [comment]
whisperX now supports the output of speaker embeddings: https://github.com/m-bain/whisperX/pull/1085

**Comment by Barabazs** [comment]
Going to close this in favor of the more detailed issue #1156
